addons/purchase_stock/models/res_partner.py

```diff
@@ -0,0 +1,31 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from datetime import timedelta
+
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    def _compute_on_time_rate(self):
```

Comment:
missing depends: add an inverse o2m partner_id on purchase order line and add it as a depends